### PR TITLE
Feat/ecs/integrate client

### PR DESCRIPTION
**Pull request by aurelenc**
### INTEGRATE CLIENT PULL REQUEST

**This Pull Request includes all the elements necessary for the integration of the client as well as the implementation of this client.**

It's composed of these features :

### Client is implemented and is able to : 

- [x] Connect to a given server
- [x] Exchange package with it 
- [x] Send & receive message
- [x] Communicate position of entities with the server

### Server is implemented and able to : 

- [x] Communicate with a single client
- [x] Communicate with multiple clients
- [x] Handle messages
- [x] Handle error of parsing message
- [x] Send Messages
- [x] Get a new connection
- [x] Manage basics action of the player (move, shoot)

More over, this pr is provided with complete and detailed documentation on the operation of each of its elements

> Message for developpers, please read few comments I added to fix small things or uncertainties.

### CODE REVIEW

**Review by Cavonstavant — APPROVED** [review]
Missing documentation like always, but ok otherwise

File: src/ecs/components/Direction.hpp [hunk 1]
```diff
@@ -15,7 +15,9 @@ namespace ecs::component
      * y represents the y-axis (vertically)
      */
     struct Direction {
+        Direction(int x, int y, bool hasMoved = 0) : x(x), y(y), hasMoved(hasMoved) {}
         int x;
         int y;
+        bool hasMoved;
```
Comment (Bootoyka) [inline src/ecs/components/Direction.hpp:21]: Not sure of the utility of the hasMoved member, please update documentation if it's wrong @aurelenc

File: src/ecs/components/EntityType.hpp [hunk 2]
```diff
@@ -11,6 +11,14 @@ namespace ecs::component
 {
     /// @brief EntityType component
     struct EntityType {
+        enum Types {
```
Comment (Bootoyka) [inline src/ecs/components/EntityType.hpp:14]: What is that ? why should we need an EntityType ? pls  explain @aurelenc

File: src/ecs/systems/ManageClientEvents.hpp [hunk 3]
```diff
@@ -60,42 +71,11 @@ namespace ecs::systems
...
-                            world.registry.addComponent<ecs::component::Drawable>(
-                                bullet, {playerPath, {5, 5, 1, 1}}); // To be removed with the server/client split
-#endif
-                        }
-                    }
-                }
+                std::cout << "SHOOOOOOOOOOOOOOOOOT" << std::endl;
+                network::Message msg;
+                msg.fill(0);
+                msg.at(0) = constant::getPacketTypeKey(ecs::constant::PacketType::PLAYER_SHOT);
```
Comment (Bootoyka) [inline src/ecs/systems/ManageClientEvents.hpp:77]: I'm not sure that you want to put an integer into an array of char

File: src/ecs/systems/client/SendDirection.hpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+** RTYPE
+** File description:
+** HandleIncomingMessages
+*/
+
+#pragma once
+
+#include <functional>
+#include <iostream>
+#include "../client/NetworkClient.hpp"
```
Comment (Bootoyka) [inline src/ecs/systems/client/SendDirection.hpp:12]: i'm sure you can do better @Cavonstavant

File: src/server/Server.cpp [hunk 5]
```diff
@@ -52,8 +51,13 @@ namespace network
         if (!error) {
```
Comment (Bootoyka) [inline src/server/Server.cpp:51]: unused bytesTransferred parameter @EmileEttel

File: src/ecs/components/EntityType.hpp (hunk 2 above)
Comment (aurelenc) [reply src/ecs/components/EntityType.hpp:14]: Required to let clients know what entity the server creates

File: src/ecs/systems/ManageClientEvents.hpp (hunk 3 above)
Comment (aurelenc) [reply src/ecs/systems/ManageClientEvents.hpp:76]: I'm sure I want

File: src/ecs/components/Direction.hpp (hunk 1 above)
Comment (aurelenc) [reply src/ecs/components/Direction.hpp:22]: Required for optimization purposes

File: src/ecs/systems/client/SendDirection.hpp (hunk 4 above)
Comment (Cavonstavant) [reply src/ecs/systems/client/SendDirection.hpp:12]: i'm on it @Bootoyka

**Pull request merged**